pkg/util/distinct_value_collector.go

```diff
@@ -8,7 +8,7 @@ type DistinctValueCollector[T comparable] struct {
 	totalLen int
 }
 
-// NewDistinctStringCollector with the given maximum data size. This is calculated
+// NewDistinctValueCollector with the given maximum data size. This is calculated
```

Comment:
i like that you removed the DistinctStringCollector and consolidated here. personally i would have just made the DistinctStringCollector a wrapper on DistinctValueCollector so all of the calling code could stay the same.

Reply:
Having to clone strings to free the underlying file made consolidating this too ugly.